src/uwtools/config/jinja2.py

```diff
@@ -182,20 +182,20 @@ def render(
     if values_needed:
         return _values_needed(undeclared_variables)
 
-    # Check for missing values required to render the template. If found, report them and raise an
-    # exception.
+    # If partial rendering has been requested, do a best-effort render. Otherwise, report any
+    # missing values and return an error to the caller.
 
-    missing = [var for var in undeclared_variables if var not in values.keys()]
-    if missing:
-        return _log_missing_values(missing)
+    if partial:
+        rendered = Environment(undefined=DebugUndefined).from_string(template_str).render(values)
+    else:
+        missing = [var for var in undeclared_variables if var not in values.keys()]
```

Comment:
Would this information be helpful as debugging info?

Reply:
Two lines below, the missing values are passed into a function that logs them. Do you have something else in mind?

But I also think we need to revisit this, because it doesn't seem to work correctly with dot-separated variable names. For example, this works fine:
```
% cat config.yaml 
roses: "{{ colors.color1 }}"
violets: "{{ colors.color2 }}"

% cat values.yaml 
colors:
  color1: red
  color2: blue

% uw template render --input-file config.yaml --values-file values.yaml
roses: "red"
violets: "blue"
```
But if I remove the `violets` line from `values.yaml`:
```
% uw template render --input-file config.yaml --values-file values.yaml
roses: "red"
violets: ""

% uw template render --input-file config.yaml --values-file values.yaml --values-needed
[2024-02-29T02:06:13]     INFO Value(s) needed to render this template are:
[2024-02-29T02:06:13]     INFO colors
```
I think our `Jinja2.undeclared_variables()` method (which the `missing` value is based on) needs to be more sophisticated. I'll open a ticket for this.

I'd like to understand the debugging idea you mentioned above, though, in case we can work it into a future PR.